### PR TITLE
Resolves #9

### LINKED ISSUE #9: Worked Through Filesystems Example Question
We should have a worked through example of pretty much every calculation that you can think of for filesystems meaning.

* Max/Min Size of a file
* Size on disk vs Size of file

Pictures would be nice.

### PR DESCRIPTION
**Pull request by aneeshdurg**
Also cleaned up some minor typos/reworded a couple sentences

### CODE REVIEW

File: filesystems/filesystems.tex [hunk 1]
```diff
@@ -251,10 +251,9 @@ \subsection{Unix Directory Conventions}
   \item \keyword{~~} is the name of the home directory usually
 \end{enumerate}
 
-Note that \keyword{...} is NOT a valid representation of any directory.
-This not the grandparent directory.
-It \emph{could} however be the name of a file on disk.
-Though confusingly, \keyword{zsh} provides this as a handy shortcut to the grandparent directory should it exist.
+Unituitively, \keyword{...} does not refer to the `grandparent' directory (parent of the parent). More generally, there are only reserved aliases for the current directory and the parent directory.
```
Comment (bhuvy2) [inline filesystems/filesystems.tex]: Typo

Unintuitively
There are -> they are

File: filesystems/filesystems.tex [hunk 2]
```diff
@@ -251,10 +251,9 @@ \subsection{Unix Directory Conventions}
...
+However, \keyword{...} \emph{could} however be the name of a file or direcotyr on disk. (You can try this with \texttt{mkdir ...}).
```
Comment (bhuvy2) [inline filesystems/filesystems.tex]: Typo on directory

File: filesystems/filesystems.tex [hunk 3]
```diff
@@ -251,10 +251,9 @@ \subsection{Unix Directory Conventions}
...
+Confusingly, the shell \keyword{zsh} does interprent \keyword{...} as a handy shortcut to the grandparent directory (should it exist) while expanding shell commands.
```
Comment (bhuvy2) [inline filesystems/filesystems.tex]: Interprent

File: filesystems/filesystems.tex [hunk 4]
```diff
@@ -1148,8 +1147,7 @@ \section{Simple Filesystem Model}
 
 \subsection{How big is our file?}
 The size of a file is not something that we can compute by staring at it. We need additional information that is
-typically stored in the inode. This is because the filesystem isn't aware of the actual contents of what is in a file.
-That data is considered the user's. However, we can compute an upper bound by looking at how many blocks the file uses.
+typically stored in the inode. This is because the filesystem isn't aware of the actual contents of what is in a file.  That data is considered the user's. However, we can compute an upper bound by looking at how many blocks the file uses.
```
Comment (bhuvy2) [inline filesystems/filesystems.tex]: We should try to break by sentences for easing editing

File: filesystems/filesystems.tex [hunk 5]
```diff
@@ -1166,6 +1164,25 @@ \subsection{How big is our file?}
...
+Note that when a user writes the first byte, a data block will be allocated. Since each data block is $4KiB$, we find that $4KiB$ is the minimum disk usage for a non-empty file.
+Here, we observe that the file size will only be $1B$, despite the fact that $4KiB$ of the disk is used -- there is a distinction between file size and disk usage because of overhead!
+
+Finding maximum is slightly more involved. As we saw earlier in this chapter, a filesystem with this structure can have $1024$ data block numbers in one
+indirect block. This implies that the maximum filesize can be $2*4KiB + 1024*4KiB = 4MiB + 8KiB$ (after accounting for the direct blocks as well).
+However, on disk we also store the indirect block itself. This means that an additional $4KiB$ of overhead will be used to account for the indirect block, so the total
+disk usage will be $4MiB + 12KiB$.
+
+Note that when only using direct blocks, completely filling up a direct block implies that our filesize and our disk usage are the same thing!
+While it would seem like we always want this idea scenario, it puts a very restrictive limit on the maximum filesize.
```
Comment (bhuvy2) [inline filesystems/filesystems.tex]: Idea -> ideal

File: filesystems/filesystems.tex (hunk 1 above)
Comment (aneeshdurg) [reply filesystems/filesystems.tex]: I meant to write "there are", maybe the wording is just bad in general - will reword it

**Pull request merged**